src/resolve-celo.ts

```diff
@@ -0,0 +1,104 @@
+import { AbiItem } from 'web3-utils'
+import { Contract } from 'web3-eth-contract'
+import { OdisUtils } from '@celo/identity'
+import { AuthSigner, ServiceContext } from '@celo/identity/lib/odis/query'
+
+import { NameResolver, NameResolutionResults, ResolutionKind } from './types'
+
+const DEFAULT_NETWORK_TIMEOUT = 5_000
+
+export class ResolveCelo implements NameResolver {
+  private federatedAttestationsContract: Contract
+  private authSigner: AuthSigner
+  private account: Address
+  private serviceContext: ServiceContext
+  private networkTimeout: number
+
+  static readonly FederatedAttestationsContractAbi: AbiItem[] =
+    require('./abis/FederatedAttestations.json').abi
+
+  // https://github.com/celo-org/SocialConnect/blob/main/protocol.md#smart-contract-addresses
+  static readonly AlfajoresFederatedAttestationsProxyContractAddress: Address =
+    '0x70F9314aF173c246669cFb0EEe79F9Cfd9C34ee3'
+  static readonly MainnetFederatedAttestationsProxyContractAddress: Address =
+    '0x0aD5b1d0C25ecF6266Dd951403723B2687d6aff2'
+
+  constructor({
+    providerUrl,
+    federatedAttestationsProxyContractAddress,
+    authSigner,
+    account,
+    serviceContext,
+    networkTimeout,
+  }: {
+    providerUrl: string
+    federatedAttestationsProxyContractAddress: Address
+    authSigner: AuthSigner
+    account: Address
+    serviceContext: ServiceContext
+    networkTimeout?: number
+  }) {
+    this.networkTimeout = networkTimeout ?? DEFAULT_NETWORK_TIMEOUT
+    const web3 = new Web3(
+      new Web3.providers.HttpProvider(providerUrl, {
+        timeout: this.networkTimeout,
+      }),
+    )
+    const federatedAttestationsContract = new web3.eth.Contract(
+      ResolveCelo.FederatedAttestationsContractAbi,
+      federatedAttestationsProxyContractAddress,
+    )
+
+    this.federatedAttestationsContract = federatedAttestationsContract
+    this.authSigner = authSigner
+    this.account = account
+    this.serviceContext = serviceContext
+  }
+
+  async resolve(id: string): Promise<NameResolutionResults> {
+    let timer: ReturnType<typeof setTimeout> | null = null
+    try {
+      const abortController = new AbortController()
+      timer = setTimeout(() => abortController.abort(), this.networkTimeout)
+      const identifier = (
+        await OdisUtils.Identifier.getObfuscatedIdentifier(
+          id,
+          OdisUtils.Identifier.IdentifierPrefix.PHONE_NUMBER,
+          this.account,
+          this.authSigner,
+          this.serviceContext,
+          undefined, // blindingFactor
+          undefined, // clientVersion
+          undefined, // blsBlindingClient
+          undefined, // sessionID
+          undefined, // keyVersion
+          undefined, // endpoint
+          abortController,
+        )
+      ).obfuscatedIdentifier
+      clearTimeout(timer)
+
+      const attestations = await this.federatedAttestationsContract.methods
+        .lookupAttestations(identifier, [this.account])
```

Comment:
Note that if you only pass in the Valora issuer account here you will only be able to see attestations published by Valora. Even if you don't want to allow cross-wallet lookups now, it might make sense to add the option to pass a list of trusted issuers either to this function or to the constructor so it's easier in the future

Reply:
This may also warrant adding a way to optionally store the issuer address on the NameResolution type

Reply:
ohhh, what's the right way to pass in the current set of "suggested" trusted issuers?

Reply:
Right now the set of issuers is so small that you can just hard code them in. We have the Kaala and Libera addresses listed in the docs [here](https://github.com/celo-org/SocialConnect) and can add Valora and Node's addresses as well. If the need arises we have ideas about how to use a smart contract (possibly a[ token curated registry](https://github.com/celo-org/SocialConnect)) to manage the "suggested" list of issuers